### PR TITLE
Functions for outputting options

### PR DESCRIPTION
**Pull request by peterschrammel**
This can be used to output the options set at the beginning
of an analysis in order to understand the settings.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [X] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by smowton** [comment]
Should probably add a simple user / test at the same time?

**Comment by martin-cs** [comment]
FWIW I'm planning on not only doing this but also dumping a check sum of the analyzer binary and the input goto-program to help make the chain of reproducibility as strong as possible.

So, +1, sounds good to me.

**Comment by peterschrammel** [comment]
Added XML support and JBMC as example user.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: f11668bd722ba2a0b9625238db24f1eccf8d65c0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87405251

**Comment by smowton** [comment]
Great, now it's easy to add a simple regression test ;)

File: jbmc/src/jbmc/jbmc_parse_options.cpp [hunk 1]
```diff
@@ -422,6 +423,24 @@ int jbmc_parse_optionst::doit()
            << "-bit " << config.this_architecture() << " "
            << config.this_operating_system() << eom;
 
+  // output the options
+  switch(ui_message_handler.get_ui())
+  {
+    case ui_message_handlert::uit::PLAIN:
+      options.output(debug());
```
Comment (tautschnig) [inline jbmc/src/jbmc/jbmc_parse_options.cpp]: I believe you could use `conditional_output(debug(), options.output);`

**Review by tautschnig — APPROVED** [review]

**Comment by peterschrammel** [comment]
@smowton, added.

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Review by smowton — APPROVED** [review]

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Comment by tautschnig** [comment]
@peterschrammel: That newly added test is failing, it seems the options are not actually printed.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 5b8d9a3446cca538878af2d2bce0b723692d2270).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88728095

**Pull request merged**